WordPress/gutenberg
# Font Library REST API: sanitize font family and font face settings

## What?

Ensures font family and font face settings are sanitized when creating or updating through the REST API endpoints.

Addresses part of https://github.com/WordPress/gutenberg/issues/58464

## Why?

- Clean and normalize font family and font face settings data.
- To mitigate possible XSS attacks.

## How?

- Uses the `sanitize_callback` param for each setting in the schema, and manually calls the callback for each property because the settings are submitted as JSON string and aren't handled by normal REST API mechanisms.
- Updates `WP_Font_Utils::format_font_family` to `WP_Font_Utils::santize_font_family` and calls `sanitize_text_field` on the font family value so it is sanitized in addition to being formatted.
- Manually removes `arg_options` from the public schema of font family and font face controllers, because the parent controller doesn't handle that for child properties.
- Add phpunit tests for the above.

## Testing Instructions

- See updated unit tests.
- Try creating a font family or font face with tags, whitespace, or line breaks in the settings values and see that they are sanitized before being created.
- Check that installing from the Google Fonts collection works as expected.
